sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs

```diff
@@ -460,6 +459,31 @@ await TransferFailedEventHandler.RaiseAsync(
                     ClientDiagnostics)
                     .ConfigureAwait(false);
             }
+
+            // Whether or not we were able to trigger the cancellation and successfully clean up
+            // we should always call CheckAndUpdateCancellationStateAsync to correctly make
+            // sure the job part is in the correct state.
+            try
+            {
+                await CheckAndUpdateCancellationStateAsync().ConfigureAwait(false);
+            }
+            catch (Exception checkUpdateException)
```

Comment:
Would this ever be because of a cancellation token? We don't want to send an event for operation cancelled or similar errors.

Reply:
Nope. I checked the code path for what this calls. Even going down the DisposeHandlersAsync and OnTransferStateChangedAsync code path, no cancellation token is passed here.